programs/drift/src/state/order_params.rs

```diff
@@ -65,35 +66,73 @@ impl OrderParams {
                 PositionDirection::Long => {
                     let ask_premium = perp_market.amm.last_ask_premium()?;
                     let est_ask = oracle_price.safe_add(ask_premium)?.cast()?;
-                    if self.price <= est_ask {
+                    if !is_oracle_offset_oracle && self.price <= est_ask
+                        || is_oracle_offset_oracle
+                            && self.oracle_price_offset.unwrap_or(0).cast::<i64>()?
+                                <= (est_ask as i64).safe_sub(oracle_price)?
+                    {
                         // if auction duration is empty and limit doesnt cross vamm premium, return early
                         return Ok(());
                     } else {
-                        let new_auction_start_price = new_auction_start_price.min(est_ask as i64);
-                        msg!(
-                            "Updating auction start price to {}",
-                            new_auction_start_price
-                        );
-                        self.auction_start_price = Some(new_auction_start_price);
-                        msg!("Updating auction end price to {}", self.price);
-                        self.auction_end_price = Some(self.price as i64);
+                        let mut new_auction_start_price =
+                            new_auction_start_price.min(est_ask as i64);
+
+                        if is_oracle_offset_oracle {
+                            new_auction_start_price =
+                                new_auction_start_price.safe_sub(oracle_price)?;
+                            msg!(
+                                "Updating oracle auction start price to {}",
+                                new_auction_start_price
+                            );
+                            self.auction_start_price = Some(new_auction_start_price);
+                            msg!("Updating oracle auction end price to {}", self.price);
+                            self.auction_end_price =
+                                Some(self.oracle_price_offset.unwrap_or(0) as i64);
+                        } else {
+                            msg!(
+                                "Updating auction start price to {}",
+                                new_auction_start_price
+                            );
+                            self.auction_start_price = Some(new_auction_start_price);
+                            msg!("Updating auction end price to {}", self.price);
+                            self.auction_end_price = Some(self.price as i64);
+                        }
                     }
                 }
                 PositionDirection::Short => {
                     let bid_discount = perp_market.amm.last_bid_discount()?;
                     let est_bid = oracle_price.safe_sub(bid_discount)?.cast()?;
-                    if self.price >= est_bid {
+                    if !is_oracle_offset_oracle && self.price >= est_bid
```

Comment:
might be easier to read if you make declare variables for oracle offset and non oracle offset cross